File: VB/WinWebSolution.Module/Order.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Persistent.BaseImpl

Namespace WinWebSolution.Module
    Public Class Order
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Private fDescription As String
        Public Property Description() As String
            Get
                Return fDescription
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Description", fDescription, value)
            End Set
        End Property
        Private fTotal As Decimal
        Public Property Total() As Decimal
            Get
                Return fTotal
            End Get
            Set(ByVal value As Decimal)
                SetPropertyValue("Total", fTotal, value)
                If (Not IsLoading) AndAlso (Not IsSaving) AndAlso Product IsNot Nothing Then
                    Product.UpdateOrdersTotal(True)
                    Product.UpdateMaximumOrder(True)
                End If
            End Set
        End Property
        Private fProduct As Product
        <Association("Product-Orders")> _
        Public Property Product() As Product
            Get
                Return fProduct
            End Get
            Set(ByVal value As Product)
                Dim oldProduct As Product = fProduct
                SetPropertyValue("Product", fProduct, value)
                If (Not IsLoading) AndAlso (Not IsSaving) AndAlso oldProduct IsNot fProduct Then
                    oldProduct = If(oldProduct, fProduct)
                    oldProduct.UpdateOrdersCount(True)
                    oldProduct.UpdateOrdersTotal(True)
                    oldProduct.UpdateMaximumOrder(True)
                End If
            End Set
        End Property
    End Class
End Namespace

File: CS/WinWebSolution.Module/Order.cs
using System;
using DevExpress.Xpo;
using DevExpress.Persistent.BaseImpl;

namespace WinWebSolution.Module {
    public class Order : BaseObject {
        public Order(Session session) : base(session) { }
        private string fDescription;
        public string Description {
            get { return fDescription; }
            set { SetPropertyValue("Description", ref fDescription, value); }
        }
        private decimal fTotal;
        public decimal Total {
            get { return fTotal; }
            set {
                SetPropertyValue("Total", ref fTotal, value);
                if (!IsLoading && Product != null) {
                    Product.UpdateOrdersTotal(true);
                    Product.UpdateMaximumOrder(true);
                }
            }
        }
        private Product fProduct;
        [Association("Product-Orders")]
        public Product Product {
            get { return fProduct; }
            set {
                Product oldProduct = fProduct;
                SetPropertyValue("Product", ref fProduct, value);
                if (!IsLoading && oldProduct != fProduct) {
                    oldProduct = oldProduct ?? fProduct;
                    oldProduct.UpdateOrdersCount(true);
                    oldProduct.UpdateOrdersTotal(true);
                    oldProduct.UpdateMaximumOrder(true);
                }
            }
        }
    }
}